Add a new README file (#1)
ubuntu-rocks/.github: profile/README.md

    
<div align="center">

>  <img align="center" src="assets/logo.gif">

Reviewer: This is using the old Ubuntu logo. I'd say it's fine for now, especially as it's saying "under construction", but should be changed at some point.

Author: it can even be removed. feel free to do so. that's just aesthetics, and not worth spending too much time on for now

 
 :construction: This project is a work in progress :construction:

 # Welcome to the Ubuntu ROCKs project! 👋 
</div>

If you're looking to build a top-shelf container image, you are in the right place.

## 🍿 Before you get started...

Here are a few things you should know before you jump in.

### What is a ROCK?

> A ROCK is a **container image**. 

Reviewer: I am fine with this language, but "container image" could also be LXC, even snaps. I wonder if this would be a good place to start referring the OCI standard (I only saw one reference to it below). Feel free to partially reject the suggestion, but I also added "meant to be used with Kubernetes" as I think this should be an important foundation for the project. We are going to make design decisions for these ROCKs and having a clear target in mind should help our focus. (Ofc, Kubernetes is used here as a _standard_ and it could be any Kubernetes-compatible platform).

```suggestion
A ROCK is a **container image**. It is compliant with the OCI standard and meant to be used with Kubernetes.
```

Author: :+1: for the OCI part, but why k8s? I wouldn't tie ROCKs to a specific container orchestration technology. You could maybe say it is Cloud Native, thus hinting that it sits on the same realm as kubernetes and friends, but saying "it is meant for k8s" will throw off single-container users (Docker, podman, etc.)

Reviewer: It's a solving-very-well-one-specific-use-case vs trying-to-address-all-of-them issue. (We can remove it from this section, but the point is that in a design conflict we should give priority to the K8s usage scenario.)

```suggestion
A ROCK is a **[container image](https://ubuntu.com/containers/what-are-containers)**. ROCKs are [cloud-native](https://github.com/cncf/toc/blob/main/DEFINITION.md) artefacts, compliant with [the OCI standard](https://github.com/opencontainers/image-spec/).
```

### How is it different from a regular Docker image?

Both are container images, and both are compatible with all de facto container runtimes and orchestration engines out there (like Docker and Kubernetes).

The big difference between the two relies mainly on the ROCK's design and supply chain.

The ROCK build system complements your typical container image build (e.g. from a Dockerfile) by offering additional steps which will make sure your final container image is:

>  - **the latest & greatest:** the ROCK profits from the underlying Ubuntu's software release updates, keeping your container images always up to date with upstream;

Reviewer: Actually, this is wrong. Both statements are true but there's no causality, or worse they could be opposites.
Ubuntu's software updates are awesome for stability and provide a huge source of stable software. However, in order to build the latest and greatest, one might want to plug their build pipeline directly on upstream's source code repository.

```suggestion
 - **the latest & greatest:** the ROCKs machinery provides the build infrastructure, flows, and publication mechanisms in order to build both _edge_ channels delivering upstream-centric releases on Day 1 and _stable_ LTS channels with long-term maintained releases and security updates;
```

 - **production-grade:** the ROCK build system only builds the container image if it is compliant with the latest standards, secure, and passes all the tests;

>  - **dependable:** new ROCKs are released with the same cadence as Ubuntu LTS releases and as soon as there are new security updates upstream;

Reviewer: ```suggestion
 - **dependable:** ROCKs' long-term maintained channels are released with a predictable cadence, at least every new Ubuntu LTS release, and are updated as soon as there are new security updates upstream;
```

 - **ubiquitous:** support for all major architectures, and publish in all major registries and marketplaces;

>  - **consistent:** the internals of the ROCK container image are predictable and consistent, as its structure is compliant with OCI and enforced at build time;

Reviewer: Note for later to create a public ROCK spec (open to community contributions) and link it here. This spec is key to ROCKs consistency.

Author: moved to a ticket

 - **performant:** ROCKs are optimised for size, startup time, speed, memory and rollout efficiency;
 - **open:** this is an open-source project, with an open design and open development;

>  - Optimise for size, startup time, speed, memory and rollout efficiency;

Reviewer: ?
```suggestion
```

 - **ruled by meritocracy:** the aim is to have field experts take responsibility for their ROCKs;

>  - **opinionated:** a ROCK offers the best user experience for the most prevalent 90% use cases;

Reviewer: ```suggestion
 - **opinionated:** a ROCK focus on the best user experience for the most prevalent use cases;
```

 - **driven by community:** fair, consistent, accountable governance of an Ubuntu community.

## 🌈 Joining the Ubuntu ROCKs project

To join the Ubuntu ROCKs project we'll have to host your GitHub repository within [this GitHub organisation](https://github.com/ubuntu-rocks).

For the moment this process is manual, so please open a new issue to initiate your onboarding process.

### Creating my first ROCK

Once your onboarding process is finalized, you'll find your application repository `<app>` hosted at `https://github.com/ubuntu-rocks/<app>`. Now, to containerize your application into a ROCK, you just need to follow these steps:

>  1. add a `rockcraft.yaml` ([template here](../rockcraft.yaml.template)) to your project's root (in the same path as the Dockerfile), 

Reviewer: broken link

Author: sry, ignore this. These last sections were meant to be commented

>  2. once you're code is ready to be built into a ROCK, create a Git branch in your repo, with the following name: `channels/<imageName:tag>`

Reviewer: I think this section needs work! Also, we need to agree on a structure for the channels to branches relation, and their matching build pipelines.

Author: yes yes :p same as above. All of this is just a placeholder for later
